src/components/MyAccount/NewSettings/EmailForm.tsx

```diff
@@ -0,0 +1,243 @@
+
+    if (firstEmailEmpty && (!value || !validateEmail(value, updatedEmails))) {
+      setError(true)
+    } else {
+      const hasInvalidEmail = updatedEmails.some(
+        (email) => !validateEmail(email, updatedEmails)
+      )
+      setError(hasInvalidEmail)
+    }
+  }
+
+  const handleRemoveEmail = (index) => {
+    const updatedEmails = tempEmails.filter((_, i) => i !== index)
+    setTempEmails(updatedEmails)
+
+    // Immediately revalidate remaining emails.
+    const hasInvalidEmail = updatedEmails.some(
+      (email) => !validateEmail(email, updatedEmails)
+    )
+    setError(hasInvalidEmail)
+  }
+
+  const handleAddEmail = () => {
+    const updatedEmails = [...tempEmails, ""]
+    setTempEmails(updatedEmails)
+
+    // Immediately revalidate all emails.
+    const hasInvalidEmail = updatedEmails.some(
+      (email) => !validateEmail(email, updatedEmails)
+    )
+    setError(hasInvalidEmail)
+  }
+
+  const handleClearableCallback = (index) => {
+    const updatedEmails = [...tempEmails]
+    updatedEmails[index] = ""
+    setTempEmails(updatedEmails)
+    setError(true)
+  }
+
+  const cancelEditing = () => {
+    setTempEmails([...emails])
+    setIsEditing(false)
+    setError(false)
+  }
+
+  const submitEmails = async () => {
+    setIsLoading(true)
+    setIsEditing(false)
+    const validEmails = tempEmails.filter((email) =>
+      validateEmail(email, tempEmails)
+    )
+    try {
+      const response = await fetch(
+        `/research/research-catalog/api/account/settings/${patronData.id}`,
+        {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ emails: validEmails }),
+        }
+      )
+
+      if (response.status === 200) {
+        await getMostUpdatedSierraAccountData()
+        setIsSuccess(true)
+      } else {
+        setIsFailure(true)
+        setTempEmails([...emails])
+      }
+    } catch (error) {
+      console.error("Error submitting emails:", error)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  return (
+    <>
+      {isLoading ? (
+        <SkeletonLoader contentSize={2} showImage={false} headingSize={0} />
+      ) : (
+        <Flex
+          flexDir={{ base: "column", lg: "row" }}
+          alignItems="flex-start"
+          width="100%"
+        >
+          <SettingsLabel icon="communicationEmail" text="Email" />
+          {isEditing ? (
+            <Flex
+              marginLeft={{ base: "l", lg: "unset" }}
+              marginTop={{ base: "xs", lg: "unset" }}
+              flexDir="column"
+              width="-webkit-fill-available"
+            >
+              {tempEmails.map((email, index) => (
+                <Form id="email-form" key={index} gap={"grid.xxs"}>
+                  <Flex mb="s" width="fill">
+                    <TextInput
+                      sx={{
+                        width: {
+                          base: "87%",
```

Comment:
Why this value? Did it just look right?

Reply:
I think it's noticeable enough that it'll be picked up. A static value sounds good but don't think it'll scale either. Let's keep this as is for now.

Reply:
you can reference the original [My Account Figma ](https://www.figma.com/design/k2E6Fc3bFV2GebWwCkmJQD/My-Account?node-id=2834-49530&node-type=canvas&t=XQqkUwOe2U2TltEc-0) file for guidelines on widths and button location. I think it's fair to assume that those should stay the same for button placement and widths.